[#15945] More descriptive server connector start logs

tristantarrant opened the pull request:
Closes #15945

jabolina commented:
@tristantarrant, I can't identify why it happens with this diff, but it is failing hard in the RESP tests for rolling upgrades. I captured the following exception:

```
14:07:16,588 ERROR [o.i.s.r.RespHandler] Command CLUSTER failed, pass to writerjava.util.concurrent.CompletionException: java.lang.NullPointerException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:636)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2194)
	at org.infinispan.manager.impl.AllClusterExecutor.lambda$submitConsumer$8(AllClusterExecutor.java:200)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at org.infinispan.remoting.transport.AbstractRequest.complete(AbstractRequest.java:69)
	at org.infinispan.remoting.transport.impl.SingleTargetRequest.onResponse(SingleTargetRequest.java:45)
	at org.infinispan.remoting.transport.impl.SingleTargetRequest.onResponse(SingleTargetRequest.java:25)
	at org.infinispan.remoting.transport.impl.RequestRepository.addResponse(RequestRepository.java:50)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processResponse(JGroupsTransport.java:1574)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processMessage(JGroupsTransport.java:1470)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport$ChannelCallbacks.up(JGroupsTransport.java:1640)
	at org.jgroups.JChannel.up(JChannel.java:678)
	at org.jgroups.stack.ProtocolStack.up(ProtocolStack.java:938)
	at org.jgroups.protocols.FRAG2.up(FRAG2.java:137)
	at org.jgroups.protocols.FlowControl.up(FlowControl.java:244)
	at org.jgroups.protocols.FlowControl.up(FlowControl.java:244)
	at org.jgroups.protocols.pbcast.GMS.up(GMS.java:799)
	at org.jgroups.protocols.pbcast.STABLE.up(STABLE.java:234)
	at org.jgroups.protocols.UNICAST3.deliverMessage(UNICAST3.java:1173)
	at org.jgroups.protocols.UNICAST3.addMessage(UNICAST3.java:898)
	at org.jgroups.protocols.UNICAST3.handleDataReceived(UNICAST3.java:880)
	at org.jgroups.protocols.UNICAST3.up(UNICAST3.java:489)
	at org.jgroups.protocols.pbcast.NAKACK2.up(NAKACK2.java:643)
	at org.jgroups.protocols.VERIFY_SUSPECT2.up(VERIFY_SUSPECT2.java:104)
	at org.jgroups.protocols.FailureDetection.up(FailureDetection.java:181)
	at org.jgroups.protocols.FD_SOCK2.up(FD_SOCK2.java:190)
	at org.jgroups.protocols.MERGE3.up(MERGE3.java:271)
	at org.jgroups.protocols.Discovery.up(Discovery.java:304)
	at org.jgroups.stack.Protocol.up(Protocol.java:360)
	at org.jgroups.protocols.TP.passMessageUp(TP.java:1127)
	at org.jgroups.util.SubmitToThreadPool$SingleMessageHandler.run(SubmitToThreadPool.java:105)
	at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(ThreadPerTaskExecutor.java:314)
	at java.base/java.lang.VirtualThread.run(VirtualThread.java:329)
Caused by: java.lang.NullPointerException
	at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at org.infinispan.server.resp.commands.cluster.SLOTS.lambda$requestNodesNetworkInformation$2(SLOTS.java:141)
	at org.infinispan.manager.impl.AllClusterExecutor.lambda$submitConsumer$5(AllClusterExecutor.java:190)
	at org.infinispan.manager.impl.AbstractClusterExecutor.consumeResponse(AbstractClusterExecutor.java:66)
	at org.infinispan.manager.impl.AllClusterExecutor.lambda$submitConsumer$7(AllClusterExecutor.java:190)
	... 33 more
```

So it seems one of the addresses is coming as null when retrieving the members.

tristantarrant commented:
I've changed things so that the internal protocol name is unchanged, and each protocol's `toString()` outputs the proper user-facing name. 
I've also cleaned up the format a bit, including the connector `name` (if specified), the socket binding and address in the details

```
INFO  [o.i.SERVER] ISPN080018: Connector RESP [address=internal, auth=RESP]
INFO  [o.i.SERVER] ISPN080018: Connector Memcached [address=internal, protocol=AUTO, auth=SCRAM-SHA-512,SCRAM-SHA-384,SCRAM-SHA-256,SCRAM-SHA-1,DIGEST-SHA-512,DIGEST-SHA-384,DIGEST-SHA-256,DIGEST-SHA,CRAM-MD5,DIGEST-MD5]
INFO  [o.i.SERVER] ISPN080018: Connector Hot Rod [name=hotrod-default, address=internal, protocol=HOTROD/4.1, auth=SCRAM-SHA-512,SCRAM-SHA-384,SCRAM-SHA-256,SCRAM-SHA-1,DIGEST-SHA-512,DIGEST-SHA-384,DIGEST-SHA-256,DIGEST-SHA,CRAM-MD5,DIGEST-MD5]
INFO  [o.i.SERVER] ISPN080018: Connector REST [name=rest-default, address=internal, auth=DIGEST]
INFO  [o.i.SERVER] ISPN005055: Using transport: Epoll
INFO  [o.i.SERVER] ISPN080018: Connector SinglePort [binding=default, address=http://127.0.0.1:11222, routes=Resp/HotRod/Memcached/REST]
```
